Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Sliders/PinchSlider.cs

```diff
@@ -33,20 +35,66 @@ public GameObject ThumbRoot
             }
         }
 
-        [Range(0, 1)]
+        #region MyValue 
+        [SerializeField]
+        private int minValue = -4;
+        [SerializeField]
+        private int maxValue = 4;
+
+        [SerializeField]
+        private bool wholeNumbers = true;
+
         [SerializeField]
+        private float startValue = 0f;
+
+        [Tooltip("Automatic placement of slider's steps.")]
+        [SerializeField]
+        private bool tickMarksAutoFill = true;
+
+        private float myValue = 0f;
+        public float MyValue
+        {
+            get { return myValue; }
+            set
+            {
+                var oldSliderValue = myValue;
+                myValue = value;
+                OnValueUpdated.Invoke(new SliderEventData(oldSliderValue, value, activePointer, this));
+            }
+        }
+
+        [Range(0, 1)]
         private float sliderValue = 0.5f;
         public float SliderValue
         {
             get { return sliderValue; }
             set
             {
                 var oldSliderValue = sliderValue;
-                sliderValue = value;
+                //Round value 
+                if (oldSliderValue != value)
+                {
+                    float temp = value * (maxValue - minValue);
+                    if (wholeNumbers == true)
+                    {
+                        int v = Mathf.RoundToInt(temp);
+                        if (MyValue != v + minValue)
+                        {
+                            MyValue = v + minValue;
+                            sliderValue = (float)v / ((float)maxValue - (float)minValue);
```

Comment:
Ah, so this is interesting - this implies that the fact that wholeNumbers also affects the underlying sliderValue as well - i.e. if "myValue" (again, probably could use a slightly more descriptive name but naming is also not my strength) is more of a "higher level concept" built on top of the underlying 0-100 slider value, it's interesting how it can affect the slider value's "deeper level" workings.

I think I'm gonna try to summarize my overall questions on this in a more meta question once I get through the rest of this review... stay tuned!